### PR TITLE
Bitcoin units

### LINKED ISSUE #96: Add Bitcoin Units to the Settings/General Tab
A radio-style control to select a bitcoin display unit should be added to the Settings/General tab.

## Functional Requirements
- [ ] the options are BTC, mBTC, μBTC and satoshi.
- [ ] the default is BTC.
- [ ] when the option is changed, any price data should be refreshed to be shown in that format (make sure cached renders, if any, don't show the old format).
- [ ] the option chosen should be shown in the format appropriate to the current language.

 ## Data
- [ ] the option chosen should be saved in local storage.

See https://github.com/OpenBazaar/OpenBazaar-Client/pull/1823 for an example of this functionality.

### PR DESCRIPTION
**Pull request by rmisio**
closes #96

### CODE REVIEW

**Comment by rmisio** [comment]
@jjeffryes This one needs a few tweaks. I'll remove the WIP form the title when it's ready.

**Comment by jjeffryes** [comment]
@rmisio gotcha.

**Comment by rmisio** [comment]
ok, @jjeffryes. This one is good to go.

File: js/views/modals/listingDetail/Listing.js [hunk 1]
```diff
@@ -361,6 +365,11 @@ export default class extends BaseModal {
       (this._$storeOwnerAvatar = this.$('.js-storeOwnerAvatar'));
   }
 
+  get $prices() {
```
Comment (jjeffryes) [inline js/views/modals/listingDetail/Listing.js]: It doesn't look like $prices is ever used?

File: js/utils/currency.js [hunk 2]
```diff
@@ -76,17 +77,28 @@ export function formatPrice(price, isBtc = false) {
...
 
-  const decimalPlaces = isBtc ? 8 : 2;
+  let convertedPrice;
 
-  return price.toFixed(decimalPlaces);
+  if (isBtc) {
+    // Format BTC price so it has up to 8 decimal places,
+    // but without any trailing zeros
+    convertedPrice =
+      parseFloat((price).toFixed(8)).toString();
```
Comment (rmisio) [inline js/utils/currency.js]: Yes, but parseFloat will also convert it back to a Number and I want it to be returned as a string here.

File: js/views/modals/Settings/General.js [hunk 3]
```diff
@@ -15,39 +16,51 @@ export default class extends baseVw {
...
-      type: 'PATCH',
-    });
+    const settingsFormData = this.getFormData();
+    this.settings.set(settingsFormData, { validate: true });
 
-    if (save) {
+    if (!this.localSettings.validationError && !this.settings.validationError) {
       const msg = {
-        msg: app.polyglot.t('settings.generalTab.statusSaving'),
+        msg: app.polyglot.t('settings.advancedTab.statusSaving'),
```
Comment (rmisio) [inline js/views/modals/Settings/General.js]: yup, good catch.

**Pull request merged**